spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/H2ServerConfiguration.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.config;
+
+import java.sql.SQLException;
+
+import org.h2.tools.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Autostart an embedded H2 database server.
+ *
+ * @author Michael Wirth
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnClass(Server.class)
+@ConditionalOnProperty(value = "spring.dataflow.embedded.database.enabled", havingValue = "true", matchIfMissing = true)
+@ConditionalOnExpression("'${spring.datasource.url:#{null}}'.startsWith('jdbc:h2:tcp://localhost:')")
+public class H2ServerConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(H2ServerConfiguration.class);
+
+	@Bean
+	public H2ServerBeanFactoryPostProcessor h2ServerBeanFactoryPostProcessor() {
+		return new H2ServerBeanFactoryPostProcessor();
+	}
+
+	@Bean
+	public Server initH2TCPServer(@Value("${spring.datasource.url}") String dataSourceUrl) {
```

Comment:
Wdyt about renaming this bean to something like `h2TcpServer`?

Reply:
I agree, I didn't want to make too many changes to the original code and didn't want to break anything in case somebody is tracking that bean.

Done